velox/functions/lib/Re2Functions.h

```diff
@@ -39,11 +39,39 @@ enum class PatternKind {
   kPrefix,
   /// Fixed pattern preceded by one or more '%', such as '%foo', '%%%hello'.
   kSuffix,
+  /// One or more fixed patterns separated by a contiguous stream of one or more
+  /// '%' e.g 'hello%%%world%', '%%hello%%%', '%%hello%%world%%'.
+  kMiddleWildcards,
   /// Patterns which do not fit any of the above types, such as 'hello_world',
   /// '_presto%'.
   kGeneric,
 };
 
+struct PatternMetadata {
+  PatternMetadata(
+      const PatternKind& patternKindArg,
+      vector_size_t numSingleWildcardsArg,
+      vector_size_t reducedPatternLengthArg,
+      vector_size_t numFixedPatternsArg,
+      const std::optional<std::vector<std::string>>& fixedPatternsArg =
+          std::nullopt)
+      : patternKind(patternKindArg),
+        numSingleWildcards(numSingleWildcardsArg),
+        reducedPatternLength(reducedPatternLengthArg),
+        numFixedPatterns(numFixedPatternsArg),
+        fixedPatterns(fixedPatternsArg) {}
+
+  const PatternKind patternKind;
+  // Number of '_' wildcards in the pattern.
+  const vector_size_t numSingleWildcards;
+  // Length of the fixed pattern in kFixed, kPrefix, and kSuffix patterns.
+  const vector_size_t reducedPatternLength;
+  // Number of fixed patterns in kMiddleWildcard patterns.
+  const vector_size_t numFixedPatterns;
```

Comment:
Why is this field needed separately ? Would we not be able to rely on the length of the fixedPatterns vector ?

Reply:
It is not really needed, when the new structure was added for the pattern metadata, I just used a separate variable for each case for clarity. I will remove it, thanks!